File: MusicService/MusicService/MessageContracts/Events/Artist.cs
using System.Collections.Generic;

namespace MusicService.MessageContracts
{
    public class Artist
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string WebsiteUrl { get; set; }
        public string FacebookUrl { get; set; }
        public string PhotoUrl { get; set; }
        public string Bio { get; set; }
        public int YearFormed { get; set; }
        public int YearBorn { get; set; }
        public string Country { get; set; }
        public string MbId { get; set; }

        public ICollection<AlbumFound> Albums { get; set; }
    }
}

File: musicsocialnetwork-client/src/models/Artist.ts
import { EventData } from './Event';
import { AlbumData } from './Album';

export interface ArtistSearchData {
  id: number;
  name: string;
  photoUrl: string;
}

export interface ArtistData {
  id: number;
  name: string;
  photoUrl: string;
  websiteUrl: string;
  facebookUrl: string;
  bio: string;
  yearFormed: number;
  yearBorn: number;
  country: string;
  genres: Array<string> | undefined;
  styles: Array<string> | undefined;
  events: Array<EventData>;
  //albums: Array<AlbumData>;
}

export interface ExtendedArtistData extends ArtistData {
  albums: Array<AlbumData>;
}

export interface ArtistDataSimple {
  id: number;
  name: string;
}

export let defaultArtistData: ArtistData = {
  id: 0,
  name: '',
  photoUrl: '',
  websiteUrl: '',
  facebookUrl: '',
  bio: '',
  yearFormed: 0,
  yearBorn: 0,
  country: '',
  genres: [],
  styles: [],
  events: []
  //albums: []
};

export let defaultArtistDataSimple: ArtistDataSimple = {
  id: 0,
  name: ''
};